framework/source/class/qx/ui/basic/Label.js

```diff
@@ -535,7 +535,12 @@ qx.Class.define("qx.ui.basic.Label",
     _applyValue : function(value, old)
     {
       // Sync with content element
-      this.getContentElement().setValue(value);
+      if (value && qx.core.Environment.get("qx.dynlocale") && value.translate) {
```

Comment:
No. I'd then go the *select* way to get rid of *get("qx.dynlocale")* to optimize the non dynlocale path. The other one isn't really optimizable at all, because you need the *if*. See updated PR.